atc/db/resource_config_scope.go

```diff
@@ -314,14 +314,26 @@ func incrementCheckOrder(tx Tx, rcsID int, version string) error {
 // Updating multiple rows using a SELECT subquery does not preserve the same
 // order for the updates, which can lead to deadlocking.
 func requestScheduleForJobsUsingResourceConfigScope(tx Tx, rcsID int) error {
-	rows, err := psql.Select("DISTINCT j.job_id").
-		From("job_inputs j").
-		Join("resources r ON r.id = j.resource_id").
-		Where(sq.Eq{
-			"r.resource_config_scope_id": rcsID,
-			"j.passed_job_id":            nil,
+	rows, err := psql.Select("DISTINCT ji.job_id").
+		From("job_inputs ji").
+		Join("resources r ON r.id = ji.resource_id").
+		Join("jobs j ON j.id = ji.job_id").
+		Where(sq.Or{
+			sq.Eq{
+				"r.resource_config_scope_id": rcsID,
+				"ji.passed_job_id":           nil,
+				"ji.trigger":                 true,
+			},
+			sq.And{
```

Comment:
why do we need this block?

Reply:
For non-trigger resource, normally new version shouldn't trigger job schedule. 

But if there is a pending build waiting for versions of the non-trigger resource in the `preparing build` phase, the trigger job schedule is necessary, otherwise it will stuck in `pending` state.